includes/gateways/paypal-standard.php

```diff
@@ -609,6 +609,19 @@ function edd_process_paypal_web_accept_and_cart( $data, $payment_id ) {
 
 	}
 
+	if( empty( $customer ) ) {
+
+		$customer = new EDD_Customer( $payment->customer_id );
+
+	}
+
+	// Record the payer email on the EDD_Customer record if it is different than the email entered on checkout
+	if( ! empty( $data['payer_email'] ) && $data['payer_email'] != $customer->email ) {
```

Comment:
This needs to be slightly changed:
1) We should be doing a `strtolower` on the email from paypal.
2) We need to search all emails for the customer, not just the primary, so we aren't trying to add an email address that we might have previously added

Keeping in line with the first item, we'll have to do an `array_map` on the emails for the customer, and do a `strtolower` and then also `strtolower` the email from paypal, that way we can be sure that we're comparing the same strings and not adding emails when they possible have different case characters and we can finish off the check with a simple `in_array` check of all the now matched-case email addresses.

Reply:
@cklosowski changes pushed.